### PR TITLE
update api to 1.8 and update readme

### PR DESCRIPTION
**Pull request by FrostML**
Update api to 1.8 and update readme for transformer and similarity_net. 
* As for transformer:
fill_constant_batch_size_like -> fill_constant

* As for similarity_net:
py_reader/PyReader -> DataLoader
fill_constant_batch_size_like -> fill_constant

### CODE REVIEW

File: PaddleNLP/machine_translation/transformer/transformer.py [hunk 1]
```diff
@@ -752,18 +750,17 @@ def beam_search():
         # caches contains states of history steps in decoder self-attention
         # and static encoder output projections in encoder-decoder attention
         # to reduce redundant computation.
+        token_shape = layers.shape(start_tokens)
```
Comment (guoshengCS) [inline PaddleNLP/machine_translation/transformer/transformer.py]: Maybe take the first dimension and replace as `batch_size`

File: PaddleNLP/machine_translation/transformer/transformer.py [hunk 2]
```diff
@@ -569,10 +569,8 @@ def transformer(model_input,
     # Padding index do not contribute to the total loss. The weights is used to
     # cancel padding index in calculating the loss.
     if label_smooth_eps:
-        # TODO: use fluid.input.one_hot after softmax_with_cross_entropy removing
-        # the enforcement that the last dimension of label must be 1.
         label = layers.label_smooth(
-            label=layers.one_hot(
+            label=fluid.one_hot(
```
Comment (guoshengCS) [inline PaddleNLP/machine_translation/transformer/transformer.py]: 还请确认下`label_smooth_eps == 0` 时，此时为`softmax_with_cross_entropy`使用hard label且使用的label输入最后一维不是1，是否会有问题。这也是之前加了TODO但是没有更新接口的原因

File: PaddleNLP/machine_translation/transformer/transformer.py (hunk 1 above)
Comment (FrostML) [reply PaddleNLP/machine_translation/transformer/transformer.py]: Done. Thanks.

File: PaddleNLP/machine_translation/transformer/transformer.py (hunk 2 above)
Comment (FrostML) [reply PaddleNLP/machine_translation/transformer/transformer.py]: 确认`softmax_with_cross_entropy`未升级，`label_smooth_eps == 0`时，会出现问题，已经 undo 静态图、动态图修改。Thanks a lot.

**Review by guoshengCS — APPROVED** [review]

**Pull request merged**